docs/source/configuration.rst

```diff
@@ -356,18 +356,42 @@ layout: :ref:`layoutconfig`.
 
 .. _templateconfig:
 
-Jinja Templating Configuration
-------------------------------
+Templating Configuration
+------------------------
 
-When thinking about Jinja templating there are two different kinds of things
-that a user might want to fill into a templated file, *variables* and
-*functions/macros*. Currently *functions* aren't implemented in any
-of the templaters.
+This section explains how to configure templating for SQL files.
 
-Variable Templating
-^^^^^^^^^^^^^^^^^^^
+When writing SQL files, users might utilise some kind of templating.
+The SQL file itself is written with placeholders which get rendered to proper
+SQL at run time.
+This can range from very simple placeholder templating to complex Jinja
+templating.
 
-Variables are available in the *jinja*, *python* and *placeholder* templaters.
+SQLFluff supports templated sections in SQL, see :ref:`templater`.
+This is achieved by the following set of operations:
+
+1. SQLFluff pre-renders the templated SQL
+2. SQLFluff applies the lint and fix operations to the rendered file
+3. SQLFluff backports the rule violations to the templated section of the SQL.
+
+SQLFluff may not have access to the fully operational templater.
+But instead uses a set of user-provided dummy values to render the template,
+see below.
```

Comment:
I think the word "operational" here might be confusing. Perhaps better would be to say:

> SQLFluff does not automatically have access to the _same environment_ as your production template setup. This means it is necessary to either provide that environment, or to provide dummy values in order to effectively render the template and still generate valid SQL.